static/js/components/CandidateList.jsx

```diff
@@ -286,6 +302,94 @@ const CandidateList = () => {
     );
   };
 
+  // Annotations filtering
+  const [tableFilterList, setTableFilterList] = useState([]);
+  const [filterListQueryStrings, setFilterListQueryStrings] = useState([]);
+
+  const filterChipToAnnotationObj = (chip) => {
+    // Convert a MuiDataTable filter list chip-formatted string to an object.
+    // Returns null for improperly formatted strings
+    const tokens = chip.split(/\s\(|\):|-/);
+    let returnObject = null;
+    switch (tokens.length) {
+      case 3:
+        returnObject = {
+          origin: tokens[1].trim(),
+          key: tokens[0].trim(),
+          value: tokens[2].trim(),
+        };
+        break;
+      case 4:
+        returnObject = {
+          origin: tokens[1].trim(),
+          key: tokens[0].trim(),
+          min: tokens[2].trim(),
+          max: tokens[3].trim(),
+        };
+        break;
+      default:
+        break;
+    }
+    return returnObject;
+  };
+
+  const filterAnnotationObjToChip = (annotationObj) => {
```

Comment:
Where are these chip-formatted strings needed? Can we just pass around objects instead, or does MUI-Datatables require chip strings?

Reply:
Uh yeah, I use these to customize the list of chips generated by MUIDataTable to display what filters are currently active - otherwise the component doesn't really know how to get that info from the autoannotation column. You can actually see an example of the chip briefly in the GIF I posted above.